include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,458 @@
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return Difference of the two angle values
+///
+////////////////////////////////////////////////////////////
+constexpr Angle& operator -=(Angle& left, Angle right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary * operator to scale an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (a number)
+///
+/// \return \a left multiplied by \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator *(Angle left, float right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary * operator to scale an angle value
+///
+/// \param left  Left operand (a number)
+/// \param right Right operand (an angle)
+///
+/// \return \a left multiplied by \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator *(float left, Angle right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary *= operator to scale/assign an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (a number)
+///
+/// \return \a left multiplied by \a right
+///
+////////////////////////////////////////////////////////////
+constexpr Angle& operator *=(Angle& left, float right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary / operator to scale an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (a number)
+///
+/// \return \a left divided by \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr Angle operator /(Angle left, float right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary /= operator to scale/assign an angle value
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (a number)
+///
+/// \return \a left divided by \a right
+///
+////////////////////////////////////////////////////////////
+constexpr Angle& operator /=(Angle& left, float right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary / operator to compute the ratio of two angle values
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return \a left divided by \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr float operator /(Angle left, Angle right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of binary % operator to compute remainder of an angle value
```

Comment:
"remainder" is technically not precise, due to how it handles negative values; I think "modulo" would fit better here.
See: https://stackoverflow.com/q/13683563

We should probably mention the semantics -- people reading this likely know what % does, but they might be interested in the handling of negative numbers. Maybe something like:
```cpp
/// Examples:
/// sf::degrees(90.f) % sf::degrees(40.f)  // approximately 10
/// sf::degrees(-90.f) % sf::degrees(40.f) // approximately 30 (not -10)
```

In addition, having right-hand side negative currently fails with an assertion. We should either document this, or define the behavior for negative `right` operands.

Reply:
Oops, I didn't write tests for negative values. I added that test case for your negative example. 

I actually didn't add the assertions you originally wrote in `positiveRemainder`. I assume SFML is fine having assertions in code that could potentially be triggered by user code? I wasn't sure what the policy was on that.

Reply:
Error handling is also one of the points where we don't have a clear convention yet.

In any case, `assert` is always better than buggy/untested results. But we can also define the behavior for negative divisors -- but we would need to think a bit about use cases and how one would typically expect things to work.

Reply:
I agree. I replaced the `assert` and removed the term "remainder" from the docs.